optuna/storages/_rdb/storage.py

```diff
@@ -863,45 +876,38 @@ def _get_trials(
-                        models.TrialModel.study_id == study_id,
-                    )
-                    .order_by(models.TrialModel.trial_id)
-                    .all()
-                )
-            except sqlalchemy_exc.OperationalError as e:
-                # Likely exceeding the number of maximum allowed variables using IN.
-                # This number differ between database dialects. For SQLite for instance, see
-                # https://www.sqlite.org/limits.html and the section describing
-                # SQLITE_MAX_VARIABLE_NUMBER.
-
-                _logger.warning(
-                    "Caught an error from sqlalchemy: {}. Falling back to a slower alternative. "
-                    "".format(str(e))
-                )
-
-                trial_models = (
-                    session.query(models.TrialModel)
-                    .options(sqlalchemy_orm.selectinload(models.TrialModel.params))
-                    .options(sqlalchemy_orm.selectinload(models.TrialModel.values))
-                    .options(sqlalchemy_orm.selectinload(models.TrialModel.user_attributes))
-                    .options(sqlalchemy_orm.selectinload(models.TrialModel.system_attributes))
-                    .options(sqlalchemy_orm.selectinload(models.TrialModel.intermediate_values))
-                    .filter(models.TrialModel.study_id == study_id)
-                    .order_by(models.TrialModel.trial_id)
-                    .all()
+            query = (
+                session.query(models.TrialModel)
+                .options(sqlalchemy_orm.selectinload(models.TrialModel.params))
+                .options(sqlalchemy_orm.selectinload(models.TrialModel.values))
+                .options(sqlalchemy_orm.selectinload(models.TrialModel.user_attributes))
+                .options(sqlalchemy_orm.selectinload(models.TrialModel.system_attributes))
+                .options(sqlalchemy_orm.selectinload(models.TrialModel.intermediate_values))
+                .filter(
+                    models.TrialModel.study_id == study_id,
                 )
-                trial_models = [t for t in trial_models if t.trial_id in trial_ids]
-
+            )
+            bind = session.bind
+            if bind is None:
+                raise ValueError('"session.bind" is None')
+            dialect = bind.dialect
+            if isinstance(dialect, sqlalchemy_dialects_mssql.dialect):
+                max_params = 2100
+            elif isinstance(dialect, sqlalchemy_dialects_mysql.dialect):
+                max_params = 65535
+            elif isinstance(dialect, sqlalchemy_dialects_oracle.dialect):
+                max_params = 1000
+            elif isinstance(dialect, sqlalchemy_dialects_postgresql.dialect):
+                max_params = 32767
+            elif isinstance(dialect, sqlalchemy_dialects_sqlite.dialect):
+                max_params = 100
+            else:
+                max_params = 100  # conservative estimate
```

Comment:
Where do these params come from? Are these params solid?
I guess we should catch the exceptions.